src/Cameras/Inputs/BaseCameraPointersInput.ts

```diff
@@ -0,0 +1,328 @@
+                        return;
+                    }
+
+                    this.doMultiTouch(
+                      pointA,
+                      pointB,
+                      previousPinchSquaredDistance,
+                      pinchSquaredDistance,
+                      previousMultiTouchPanPosition,
+                      multiTouchPanPosition);
+
+                    previousMultiTouchPanPosition = {x: (pointA.x + pointB.x) / 2,
+                                                     y: (pointA.y + pointB.y) / 2,
+                                                     pointerId: evt.pointerId,
+                                                     type: p.type};
+                    previousPinchSquaredDistance = pinchSquaredDistance;
+                }
+            }
+        };
+
+        this._observer = this.camera.getScene().onPointerObservable.add(
+            this._pointerInput,
+            PointerEventTypes.POINTERDOWN | PointerEventTypes.POINTERUP |
+            PointerEventTypes.POINTERMOVE);
+
+        this._onLostFocus = () => {
+            this.log("FollowCameraPointersInput._onLostFocus");
+            pointA = pointB = null;
+        };
+
+        element.addEventListener("contextmenu",
+            <EventListener>this.onContextMenu.bind(this), false);
+        element.addEventListener("MSPointerDown",
+            <EventListener>this.onGestureStart.bind(this), false);
+        element.addEventListener("MSGestureChange",
+            <EventListener>this.onGesture.bind(this), false);
+
+        Tools.RegisterTopRootEvents([
+            { name: "blur", handler: this._onLostFocus }
+        ]);
+    }
+
+    /**
+     * Detach the current controls from the specified dom element.
+     * @param element Defines the element to stop listening the inputs from
+     */
+    public detachControl(element: Nullable<HTMLElement>): void {
+        if (this._onLostFocus) {
+            Tools.UnregisterTopRootEvents([
+                { name: "blur", handler: this._onLostFocus }
+            ]);
+        }
+
+        if (element && this._observer) {
+            this.camera.getScene().onPointerObservable.remove(this._observer);
+            this._observer = null;
+
+            if (this.onContextMenu) {
+                element.removeEventListener("contextmenu", this.onContextMenu);
+            }
+
+            if (this.onGestureStart) {
+                element.removeEventListener("MSPointerDown", <EventListener>this.onGestureStart);
+            }
+
+            if (this.onGesture) {
+                element.removeEventListener("MSGestureChange", <EventListener>this.onGesture);
+            }
+
+            this._MSGestureHandler = null;
+            this._onLostFocus = null;
+        }
+    }
+
+    /**
+     * Gets the class name of the current input.
+     * @returns the class name
+     */
+    public getClassName(): string {
+        return "ArcRotateCameraPointersInput";
```

Comment:
need to be fixed with correct name

Reply:
o yes.
Let's make `getClassName()` `abstract` in the Base class to force the user to set it in the derived class.

Reply:
hmm, interesting...
it appears `this` in `BaseCameraPointersInput` is not able to access the derived `getClassName()`.

Learning about JS internals now. stay tuned.

Reply:
Ok, setting a variable as `abstract` and returning that in the base `getClassName()` seems like a straight forward way to do this.

Interested to hear alternatives.